### ISSUE TITLE
Prepare for Angular 2

### ISSUE THREAD

**Issue by martinleopold** [StudioProcess/rvp#10]
https://angular.io/docs/ts/latest/guide/upgrade.html

- [ ] follow angular 1 style guide
  * one component (controller, component, service, filter) per file
  * folders-by-feature structure
- [x] use a module loader with ES6
- [ ] turn services and controllers into classes
- [ ] use component directives with ng 1.5 component API
```
export const heroDetail = {
  bindings: {
    hero: '=',
    deleted: '&'
  },
  template: `
    <h2>{{$ctrl.hero.name}} details!</h2>
    <div><label>id: </label>{{$ctrl.hero.id}}</div>
    <button ng-click="$ctrl.onDelete()">Delete</button>
  `,
  controller: function() {
    this.onDelete = () => {
      this.deleted(this.hero);
    };
  }
};
```
- [x] [migrate to typescript](https://angular.io/docs/ts/latest/guide/upgrade.html#!#migrating-to-typescript)

**Comment by martinleopold** [comment]
component directives and especially the component app structure seem complicated in ng 1.5: https://code.angularjs.org/1.5.6/docs/guide/component

why not go all the way and move to https://github.com/angular/angular ?
* learn ng2 concepts (which are supposedly much less than 1.5)
* learn ts
* move to .ts (quick fix: tsify instead of babelify, later: gulp-typescript)

**Comment by martinleopold** [comment]
seems tsify is buggy: https://github.com/TypeStrong/tsify/issues/147
so, either:
* don’t use tsconfig.json, or
* change bundle process to gulp-typescript (which is more work)

**Comment by martinleopold** [comment]
port to ng2:
- [ ] typescript basic setup (tsify)
- [ ] ng2 basic setup
- [ ] proper typescript gulp task
- [ ] proper build gulp task

**Comment by martinleopold** [comment]
closed in favor of #15
